nvflare/app_opt/pt/job_config/base_fed_job.py

```diff
@@ -37,78 +68,46 @@ def __init__(
-            analytics_receiver (bool | AnalyticsReceiver | None, optional): Receive analytics.
-                If not provided, a TBAnalyticsReceiver will be configured.
-            model_persistor (ModelPersistor | None, optional): how to persistor the model.
-            model_locator (ModelLocator | None, optional): how to locate the model.
-        """
+        # Call the unified BaseFedJob with PyTorch-specific settings
         super().__init__(
+            initial_model=initial_model,
             name=name,
             min_clients=min_clients,
             mandatory_clients=mandatory_clients,
+            key_metric=key_metric,
+            validation_json_generator=validation_json_generator,
+            model_selector=model_selector,
+            convert_to_fed_event=convert_to_fed_event,
+            analytics_receiver=analytics_receiver,
+            model_persistor=model_persistor,
+            framework=FrameworkType.PYTORCH,
         )
 
-        self.initial_model = initial_model
-        self.comp_ids = {}
-
-        if validation_json_generator:
-            validate_object_for_job("validation_json_generator", validation_json_generator, ValidationJsonGenerator)
-        else:
-            validation_json_generator = ValidationJsonGenerator()
-        self.to_server(id="json_generator", obj=validation_json_generator)
-
-        if intime_model_selector:
-            validate_object_for_job("intime_model_selector", intime_model_selector, IntimeModelSelector)
-            self.to_server(id="model_selector", obj=intime_model_selector)
-        elif key_metric:
-            self.to_server(id="model_selector", obj=IntimeModelSelector(key_metric=key_metric))
-
-        if convert_to_fed_event:
-            validate_object_for_job("convert_to_fed_event", convert_to_fed_event, ConvertToFedEvent)
-        else:
-            convert_to_fed_event = ConvertToFedEvent(events_to_convert=[ANALYTIC_EVENT_TYPE])
-        self.convert_to_fed_event = convert_to_fed_event
+        # PyTorch-specific model setup
+        if initial_model is not None:
+            self._setup_pytorch_model(initial_model, model_persistor, model_locator)
 
-        if analytics_receiver:
-            validate_object_for_job("analytics_receiver", analytics_receiver, AnalyticsReceiver)
-        else:
-            analytics_receiver = TBAnalyticsReceiver()
-
-        self.to_server(
-            id="receiver",
-            obj=analytics_receiver,
-        )
-
-        if initial_model:
-            self.comp_ids.update(
-                self.to_server(PTModel(model=initial_model, persistor=model_persistor, locator=model_locator))
-            )
+    def _setup_pytorch_model(
+        self,
+        model: nn.Module,
+        persistor: Optional[ModelPersistor] = None,
+        locator: Optional[ModelLocator] = None,
+    ):
+        """Setup PyTorch model with persistor and locator."""
+        from nvflare.app_opt.pt.job_config.model import PTModel
```

Comment:
suggest move this import up

Reply:
this is so it is only imported if initial_model is not None

we can move it but keeping it here can be a small performance optimization?